src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -209,34 +225,114 @@ private void loadConfig(Map<String, Object> config, AmazonCloudWatch client) {
+          }
         }
 
-        loadConfig(rules, client);
+        loadConfig(rules, cloudWatchClient, taggingClient);
     }
 
-    private void loadConfig(ArrayList<MetricRule> rules, AmazonCloudWatch client) {
+    private void loadConfig(ArrayList<MetricRule> rules, AmazonCloudWatch cloudWatchClient, AWSResourceGroupsTaggingAPI taggingClient) {
         synchronized (activeConfig) {
-            activeConfig.client = client;
+            activeConfig.cloudWatchClient = cloudWatchClient;
+            activeConfig.taggingClient = taggingClient;
             activeConfig.rules = rules;
         }
     }
 
+    private AWSCredentialsProvider getRoleCredentialProvider(Map<String, Object> config) {
+        STSAssumeRoleSessionCredentialsProvider credentialsProvider = new STSAssumeRoleSessionCredentialsProvider.Builder(
+                (String) config.get("role_arn"),
+                "cloudwatch_exporter"
+              ).build();
+        return credentialsProvider;
+    }
+    
+    private Region getRegion(Map<String, Object> config) {
+        Region region = RegionUtils.getRegion((String) config.get("region"));
+        if (region == null) {
+          region = Regions.getCurrentRegion();
+          if (region == null) {
+            throw new IllegalArgumentException("No region provided and EC2 metadata failed");
+          }
+        }
+        return region;
+    }
+    
     public String getMonitoringEndpoint(Region region) {
-      return "https://" + region.getServiceEndpoint("monitoring");
+        return "https://" + region.getServiceEndpoint("monitoring");
     }
 
-    private List<List<Dimension>> getDimensions(MetricRule rule, AmazonCloudWatch client) {
+    private List<ResourceTagMapping> getResourceTagMappings(MetricRule rule, AWSResourceGroupsTaggingAPI taggingClient) {
+	List<ResourceTagMapping> resourceTagMappings = new ArrayList<ResourceTagMapping>();
+	if(rule.awsTagSelect == null) {
+            return resourceTagMappings;
+        }
+	
+        List<TagFilter> tagFilters = new ArrayList<TagFilter>();
+        for (Map.Entry<String, List<String>> entry : rule.awsTagSelect.tagSelections.entrySet()) {
+          tagFilters.add(new TagFilter().withKey(entry.getKey()).withValues(entry.getValue()));
+        }
+
+        GetResourcesRequest request = new GetResourcesRequest().withTagFilters(tagFilters).withResourceTypeFilters(rule.awsTagSelect.resourceTypeSelection);
+        String paginationToken = "";
+        do {
+          request.setPaginationToken(paginationToken);
+          
+          GetResourcesResult result = taggingClient.getResources(request);
+          resourceTagMappings.addAll(result.getResourceTagMappingList());
+          
+          paginationToken = result.getPaginationToken();
+        } while (paginationToken != null && paginationToken != "");
+     
+        return resourceTagMappings;
+    }
+    
+    private List<String> extractResourceIds(List<ResourceTagMapping> resourceTagMappings) {
```

Comment:
name is confusing with next method, would call it getResourceARNList

Reply:
@vincentcreusotfx it is not getting ARN list, it is extracting resource IDs. But I think it was confusing because having 2 methods for that was overkill. I merged them